### ISSUE TITLE
feat(Φ7): CPU training benchmark — real BPB metrics and timing data

### ISSUE THREAD

**Issue by gHashTag** [gHashTag/trios#35]
## Goal

Run a complete 1000-step CPU training benchmark on IGLA-GF16 to obtain empirical BPB metrics, wall-clock timing, and checkpoint size for Parameter Golf submission.

## Current Status

- `trios-train-cpu` infrastructure: ✅ complete (39 tests green, clippy clean)
- Training loop: ✅ implemented with BPB logging every 34 steps
- Missing: **actual training run data**

### Hardware
- Run on Apple M2 (or equivalent ARM64 CPU)
- No GPU acceleration

### Configuration
```rust
max_steps: 1000
batch_size: 4
seq_len: 128
learning_rate: 0.001
warmup_steps: 21 (Fib #7)
grad_clip: 0.618 (phi^-1)
log_every: 34 (Fib #8)
vocab_size: 32000
```

### Required Output

1. **Training trace** (saved to file):
   - Step number (every 34 steps)
   - Loss value
   - BPB value
   - Milliseconds per step
   - Learning rate

2. **Final metrics**:
   - Final BPB
   - Total wall-clock time
   - Average ms/step
   - Checkpoint file size

3. **Plots** (optional but recommended):
   - Loss vs step
   - BPB vs step
   - Learning rate schedule

## Acceptance Criteria

- [ ] Complete 1000-step training run on CPU
- [ ] Save training trace to `results/cpu_bench_YYYYMMDD_HHMMSS.json`
- [ ] Final BPB recorded (target: < 2.0)
- [ ] Wall-clock time recorded (target: < 10 minutes for 1000 steps)
- [ ] Checkpoint file saved with size logged
- [ ] `cargo test -p trios-train-cpu` — all tests green
- [ ] NO .sh files
- [ ] All code and comments in English
- [ ] Commit with `refs #35`

### JSON Output Format
\`\`\`json
{
  "run_id": "cpu_bench_20260420_120000",
  "timestamp": "2026-04-20T12:00:00Z",
  "config": {
    "max_steps": 1000,
    "batch_size": 4,
    "seq_len": 128,
    "learning_rate": 0.001,
    "warmup_steps": 21,
    "grad_clip": 0.618,
    "log_every": 34
  },
  "metrics": {
    "final_bpb": 1.85,
    "final_loss": 1.28,
    "total_time_seconds": 245.3,
    "avg_ms_per_step": 245.3,
    "checkpoint_size_bytes": 19700000
  },
  "trace": [
    {"step": 0, "loss": 10.5, "bpb": 15.2, "ms_per_step": 280, "lr": 0.0},
    {"step": 34, "loss": 8.2, "bpb": 11.8, "ms_per_step": 250, "lr": 0.000476},
    ...
  ]
}
\`\`\`

### Binary Enhancement
Update `src/bin/train_cpu.rs` to:
1. Save trace to JSON file
2. Print summary at end
3. Measure actual checkpoint size

## Dependencies

- #32 trios-train-cpu (baseline implementation)
- #34 size optimization (follow-up, after benchmark)

## Notes

- This run establishes the empirical baseline for IGLA-GF16 CPU training
- Results will be used in PhD thesis, Zenodo paper, and Parameter Golf narrative
- Size optimization (tied embeddings, vocab pruning) will follow based on actual checkpoint size

**Comment by gHashTag** [comment]
Parameter Golf competition deadline was April 30, 2026. Competition ended. Closing.
